### ISSUE TITLE
Uncaught (in promise) Error: All values in axis param must be in range [-1, 1) but got axis 1

### ISSUE THREAD

**Issue by mariosplen** [Hyuto/yolov8-tfjs#6]
1) I added my custom model:
![image](https://github.com/Hyuto/yolov8-tfjs/assets/55495824/a2ea0762-8e5c-4c18-817f-ebb6ffcad561)
2) changed the name in app.jsx
3) and changed the labels.json accordingly:
![image](https://github.com/Hyuto/yolov8-tfjs/assets/55495824/70662a8a-60f5-4f3a-a4bc-9eab9432b4d0)

I am getting:
![image](https://github.com/Hyuto/yolov8-tfjs/assets/55495824/5fd1a204-654c-497f-b95d-cda82bc216a6)

Maybe the issue is because I have only 1 class on my model?

**Comment by Hyuto** [comment]
Hello, Sorry for late reply. 

I don't think having 1 class is the issue here. Can you provide the link to download the model (with `labels.json`)?

**Comment by mariosplen** [comment]
On the `labels.json` file, I have removed all classes and kept only one class ("box") since I trained the model from scratch.

Did you mean the `model.json` file?

I have shared all the outputs I obtained when converting to tensorflowjs, including the `web_model` and the `model.json`, in this 
[Google Drive Link](https://drive.google.com/drive/folders/1DeRkqdO6dRrIzzJrE5SzAL9zRRa8yGVN?usp=drive_link)

The export from .pt to web_model seems to finish successfully. I'm using the following code:

```
model = YOLO("best.pt")
model.export(format="tfjs")
```

I also tried the model using ONNX and it works fine. 
If you could provide some assistance, it would be greatly appreciated. 
Thank you.

**Comment by Hyuto** [comment]
Hello, thanks to provide the model for testing purpose.

I've found the issue here is on 

https://github.com/Hyuto/yolov8-tfjs/blob/b6a8bc691c0c1897802373d7da65dc889af1f451/src/utils/detect.js#L76

Before it none was pass to `squeeze` so it squeeze all of the size 1 axis. In your case the model only have 1 class so the score tensor shape would be [1, N, 1] and squeezed to [N], but we need the tensor in shape of [N, nclass] so the code break. If the model have more than 1 class that would be fine since `squeeze` operators only remove axis with size of 1.

I've fixed that issue by passing 0 (batch) as the axis need to be squeezed since we need to remove that for NMS operators.
